### PR TITLE
Ensure we have a valid console ready after every select_console switch

### PR DESCRIPTION
**Pull request by okurz**
Using the new callback from the 'select_console' as added by
https://github.com/os-autoinst/os-autoinst/pull/738 and
https://github.com/os-autoinst/os-autoinst/pull/740
we can replace the already existing explicit places for checking that a
console is ready after switching and do it properly every time except for the
specific cases where we do not want to check or do it differently.

Needs the corresponding os-autoinst changes as well as needle updates.

Needle changes:
* https://gitlab.suse.de/openqa/os-autoinst-needles-sles/merge_requests/334
* https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/173

Verification runs:
* sle: http://lord.arch/tests/6068
* sle maintenance: http://lord.arch/tests/6069
* tw: http://lord.arch/tests/6067

Related progress issue: https://progress.opensuse.org/issues/14826

### CODE REVIEW

**Comment by okurz** [comment]
you are invited to review but I would like to test more before we merge. The validation runs might also fail because they are not completed yet

File: lib/susedistribution.pm [hunk 1]
```diff
@@ -190,15 +190,16 @@ sub script_sudo($$) {
...
     my ($self, $user) = @_;
     $user ||= $testapi::username;
     if ($user eq 'root') {
         # set standard root prompt
-        type_string "PS1='# '\n";
+        type_string "PS1=\"\$(tput bold 2; tput setaf 1)#\$(tput sgr0) \"\n";
     }
     else {
-        type_string "PS1='\$ '\n";
+        type_string "PS1=\"\$(tput bold 2; tput setaf 1)\\\$\$(tput sgr0) \"\n";
```
Comment (coolo) [inline lib/susedistribution.pm]: I thought the idea was to colour only one?

**Comment by coolo** [comment]
Looking at your changes, we might just want to skip 'x11' in console_selected. I don't really like this no_callback = 1 - it's just bad API.

I think the better API is to have optional arguments to select_console that are passed to the callback as is. Then select_console('x11') would just never assert_screen and select_console('root-console', await_prompt => 0) would be specific to our callback.

File: lib/susedistribution.pm (hunk 1 above)
Comment (okurz) [inline lib/susedistribution.pm]: no. I just brought back the default color from SUSE/openSUSE to make the prompt distinguishable from the entries. The character '$' and '#' are IMHO distinguishable enough unless someone tries to cover them in a full screen needle but no single character will be different enough in that context.

**Comment by okurz** [comment]
> I think the better API is to have optional arguments to select_console that are passed to the callback as is. Then select_console('x11') would just never assert_screen and select_console('root-console', await_prompt => 0) would be specific to our callback.

sure, I thought about that as an alternative. So I would unconditionally call the callback with all additional arguments that have been supplied to `select_console` and then we can do something distribution specific in there, right?

**Comment by coolo** [comment]
That would be the idea

**Comment by okurz** [comment]
how do you get these notifications this fast?

**Comment by coolo** [comment]
which?

**Comment by okurz** [comment]
that I wrote something or do you poll that often?

**Comment by coolo** [comment]
well, github has this little bell - and it's a push event.

**Comment by okurz** [comment]
sure, if you have the window open. So you look at the screen then?

**Comment by coolo** [comment]
That's what the screen is for! :)

**Comment by okurz** [comment]
I consider myself done here for now. Updated verification runs. There are failed modules which are unrelated except for the TW build which includes a trivial needle failure that is part of the updated needle PR so *done*. I will continue to test anyway and would monitor closely failures on the production instance which would seem related.

**Comment by okurz** [comment]
sle-12-SP3-Server-DVD-HA-x86_64-Build0098@0290-fix_select_console_sle11sp4_ha_migration@64bit: http://lord.arch.suse.de/tests/6072

**Pull request merged**